### PR TITLE
Release preparations

### PR DESCRIPTION
**Pull request by hannesm**
I included benchmarks (from bench/speed) in the changes notes - results are "as expected": 
- the getrandom/getentropy is slower than a fortuna seeded only once (at least on my FreeBSD system)
- public key cryptography pays ~25% for powm_sec
- symmetric crypto and hashes are as nocrypto-master

please read carefully and comment if I forgot anything. thanks for your time. //cc @emillon @cfcs @dinosaure @avsm

### CODE REVIEW

File: bench/speed.ml [hunk 1]
```diff
@@ -48,10 +48,360 @@ let throughput title f =
...
+   "ffdhe2048",Mirage_crypto_pk.Dh.Group.ffdhe2048;
+   "ffdhe3072",Mirage_crypto_pk.Dh.Group.ffdhe3072;
+   "ffdhe4096",Mirage_crypto_pk.Dh.Group.ffdhe4096;
+   "ffdhe6144",Mirage_crypto_pk.Dh.Group.ffdhe6144]
+
+type s = { x : Z.t }
+
+let dh_secrets =
+  List.combine dh_groups @@
+  List.map (fun (se, sh) -> Obj.magic {x = Z.of_string se}, Cstruct.of_hex sh)
```
[the remark this answers is not in the document]
Comment (hannesm) [reply bench/speed.ml]: the `Mirage_crypto_pk.Dh.secret` type (` = { x : Z.t }`) is kept private, this was the easiest i could come up with

[the remark this answers is not in the document]
Comment (dinosaure) [reply bench/speed.ml]: Can we add something like `unsafe_secret` and hide it from the documentation?

[the remark this answers is not in the document]
Comment (hannesm) [reply bench/speed.ml]: no, but I'll revise the code to be Obj.magic-free.

**Comment by hannesm** [comment]
I just pushed a Obj.magic-free speed.ml; and also added checks in dh.ml validating the secret (it may be a good idea in the future to couple secret and group (in a tuple/pair), this way we could safely get rid of these checks).

**Comment by hannesm** [comment]
macOS CI fails due to timeout, the change requested by @cfcs was addressed, merging

**Pull request merged**